src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.String.cs

```diff
@@ -72,9 +100,193 @@ public bool TryReadTextString(Span<char> destination, out int charsWritten)
+
+            if (concatenatedBufferSize > destination.Length)
+            {
+                bytesWritten = 0;
+                return false;
+            }
+
+            ReadOnlySpan<byte> source = _buffer.Span;
+
+            foreach ((int o, int l) in ranges)
+            {
+                source.Slice(o, l).CopyTo(destination);
+                destination = destination.Slice(l);
+            }
+
+            bytesWritten = concatenatedBufferSize;
+            AdvanceBuffer(encodingLength);
+            DecrementRemainingItemCount();
+            return true;
+        }
+
+        private bool TryReadChunkedTextStringConcatenated(Span<char> destination, out int charsWritten)
+        {
+            List<(int offset, int length)> ranges = ReadChunkedStringRanges(CborMajorType.TextString, out int encodingLength, out int _);
+            ReadOnlySpan<byte> buffer = _buffer.Span;
+
+            int concatenatedStringSize = 0;
+            foreach ((int o, int l) in ranges)
+            {
+                concatenatedStringSize += s_utf8Encoding.GetCharCount(buffer.Slice(o, l));
+            }
+
+            if (concatenatedStringSize > destination.Length)
+            {
+                charsWritten = 0;
+                return false;
+            }
+
+            foreach ((int o, int l) in ranges)
+            {
+                s_utf8Encoding.GetChars(buffer.Slice(o, l), destination);
+                destination = destination.Slice(l);
+            }
+
+            charsWritten = concatenatedStringSize;
+            AdvanceBuffer(encodingLength);
+            DecrementRemainingItemCount();
+            return true;
+        }
+
+        private byte[] ReadChunkedByteStringConcatenated()
+        {
+            List<(int offset, int length)> ranges = ReadChunkedStringRanges(CborMajorType.ByteString, out int encodingLength, out int concatenatedBufferSize);
+            var output = new byte[concatenatedBufferSize];
+
+            ReadOnlySpan<byte> source = _buffer.Span;
+            Span<byte> target = output;
+
+            foreach ((int o, int l) in ranges)
+            {
+                source.Slice(o, l).CopyTo(target);
+                target = target.Slice(l);
+            }
+
+            Debug.Assert(target.IsEmpty);
+            AdvanceBuffer(encodingLength);
+            DecrementRemainingItemCount();
+            return output;
+        }
+
+        private string ReadChunkedTextStringConcatenated()
+        {
+            List<(int offset, int length)> ranges = ReadChunkedStringRanges(CborMajorType.TextString, out int encodingLength, out int concatenatedBufferSize);
+            ReadOnlySpan<byte> buffer = _buffer.Span;
+            int concatenatedStringSize = 0;
+
+            foreach ((int o, int l) in ranges)
+            {
+                concatenatedStringSize += s_utf8Encoding.GetCharCount(buffer.Slice(o, l));
+            }
+
+            var output = new char[concatenatedStringSize];
```

Comment:
Instead of creating a temporary `char[]`, 

```C#
        string.Create(concatenatedStringSize, ranges, (target, sources) =>
        {
            foreach ((int o, int l) in sources)
            {
                s_utf8Encoding.GetChars(buffer.Slice(o, l), target);
                target = target.Slice(l);
            }

            Debug.Assert(target.IsEmpty);
        });

Reply:
Done.